ambry-commons/src/main/java/com.github.ambry.commons/ByteBufferReadableStreamChannel.java

```diff
@@ -30,16 +36,25 @@
 public class ByteBufferReadableStreamChannel implements ReadableStreamChannel {
   private final AtomicBoolean channelOpen = new AtomicBoolean(true);
   private final AtomicBoolean channelEmptied = new AtomicBoolean(false);
-  private final ByteBuffer buffer;
+  private final List<ByteBuffer> buffers;
   private final int size;
 
   /**
    * Constructs a {@link ReadableStreamChannel} whose read operations return data from the provided {@code buffer}.
    * @param buffer the {@link ByteBuffer} that is used to retrieve data from on invocation of read operations.
    */
-  public ByteBufferReadableStreamChannel(ByteBuffer buffer) {
-    this.buffer = buffer;
-    size = buffer.remaining();
+  public ByteBufferReadableStreamChannel(ByteBuffer buffer) { this(Arrays.asList(buffer)); }
+
+  /**
+   * Constructs a {@link ReadableStreamChannel} whose read operations return data from the provided {@code bufferList}.
+   * @param bufferList list of {@link ByteBuffer} that is used to retrieve data from on invocation of read operations.
+   */
+  public ByteBufferReadableStreamChannel(List<ByteBuffer> bufferList) {
+    if (bufferList == null || bufferList.isEmpty()) {
+      throw new IllegalArgumentException("Input List of ByteBuffer cannot be null/empty for ByteBufferReadableStreamChannel");
+    }
+    buffers = new ArrayList<>(bufferList);
```

Comment:
Yes, but notice that in `MockRestRequest`, buffers can be added after construction (which cannot be done here) - so assuming the list is mutable is fair in `MockRestRequest`.

Aside from that, it might not even be important to `remove()` in `MockRestRequest`. It was probably done for a use-case that may no longer be true. When you encounter situations like this, it is best to think about whether it makes sense for this use case (and for the future). Looking at it from that point of view, even if this class supported adding buffers in the future, `remove()` may not be required. If it is, then assuming a mutable list would be a fair thing to do.